src/crypto/crypto_prover.rs

```diff
@@ -119,9 +119,14 @@ impl CryptoProver {
         let crypto_proof_request =
             BbsVerifier::new_proof_request(&sub_proof_request.revealed_attributes.as_slice(), &pk)
                 .map_err(|e| format!("could not create new proof request; {}", &e))?;
+        let required_reveal_statements = credential_signature.required_reveal_statements.to_owned();
+        let revealed_statements = sub_proof_request.revealed_attributes.to_owned();
+        let all_revealed_statements = concat_required_and_reveal_statements(
+            required_reveal_statements,
+            revealed_statements,
+        )?;
 
-        let indices: HashSet<usize> =
-            HashSet::from_iter(sub_proof_request.revealed_attributes.iter().cloned());
+        let indices: HashSet<usize> = HashSet::from_iter(all_revealed_statements.iter().cloned());
```

Comment:
As we defined `all_revealed_statements` only for the indices and don't seem to need it afterwards, we could also just pass it as is (the clone here was only to avoid the partial move, I think. ^^)

```suggestion
        let indices: HashSet<usize> = HashSet::from_iter(all_revealed_statements);
```